torch/_inductor/compile_fx_async.py

```diff
@@ -236,14 +235,19 @@ def __call__(self, args: Sequence[Any]) -> Any:
         return res
 
     def _check_and_switch_progression(self) -> None:
-        # Check if any newer progression stage is ready (in order from latest to earliest)
-        for i in range(
-            len(self._progression_futures) - 1, self._current_progression_index, -1
-        ):
-            future = self._progression_futures[i]
-            if self._post_compile_data and future and future.done():
-                self._switch_to_progression_stage(i)
-                break
+        if not self._progression_futures:
+            return
+
+        stage_index = -1
+        for i, future in enumerate(self._progression_futures):
```

Comment:
nit: This is fine. In the future if we find that the futures list is really long we could instead attach an add_done_callback() to the futures to set a flag so we can check whether it's worth checking or not.  But I suspect we'll never get to the point where that's worth it because we only do this once per compiled graph run and I expect the list to be fairly short (3-4 entries max)